### ISSUE TITLE
False positive: assignment expression with `is not None` does not narrow RHS

### ISSUE THREAD

**Issue by hsusul** [microsoft/pyright#11563]
**Describe the bug**
When an assignment expression (walrus operator) is used with `is not None` / `is None` / `!= None` / `== None`, Pyright narrows the assignment target but does **not** narrow the RHS expression. Truthiness narrowing of the same assignment expression already narrows both the target and the RHS (see existing coverage in `typeNarrowing4.py`).

This produces a false positive on valid code that is accepted when written with an equivalent truthiness check.

**Code or Screenshots**

```python
class C:
    def method1(self) -> None:
        pass

def f(b: C | None) -> None:
    # Truthiness: both `c` and `b` are narrowed to C — OK
    if c := b:
        b.method1()

    # is not None: `d` is narrowed, but `b` is not — false positive
    if (d := b) is not None:
        b.method1()  # reportOptionalMemberAccess
```

Expected: after `(d := b) is not None`, `b` should be narrowed to `C`, consistent with truthiness narrowing of `c := b`.

Actual (Pyright 1.1.411 / upstream `main`):

```text
error: "method1" is not a known attribute of "None" (reportOptionalMemberAccess)
```

**Root-cause hypothesis**
In `typeGuards.ts`, the `is None` / `is not None` path peels an assignment expression down to only its target name before matching the reference. Truthiness narrowing goes through `getTypeNarrowingCallbackForAssignmentExpression`, which also considers the RHS.

**VS Code extension or command-line**
Command-line pyright 1.1.411 (built from current `main`).

I intend to open a PR with a minimal fix and regression tests.
